### PR TITLE
fix(install_openadapt.ps1): make the git detection logic more robust and some minor refactoring

### PR DESCRIPTION
**Pull request by ayewo**
**What kind of change does this PR introduce?**
* make the check that the installer performs prior to re-installing git more robust
* minor refactoring for vars and function calls 

**Summary**
Closes https://github.com/OpenAdaptAI/OpenAdapt/issues/448

**Checklist**
* [x] My code follows the style guidelines of OpenAdapt
* [x] I have performed a self-review of my code
* [ ] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [ ] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**

### CODE REVIEW

File: install/install_openadapt.ps1 [hunk 1]
```diff
@@ -181,15 +193,15 @@ function GetPythonCMD {
         $res = Invoke-Expression "python -V"
         $versionString = $res.Split(' ')[-1]
 
-        if (ComparePythonVersion $versionString  $pythonMaxVersion) {
+        if (ComparePythonVersion $versionString) {
             return $pythonCmd
         }
-    }
+    }
```
Comment (KrishPatel13) [inline install/install_openadapt.ps1:199]: Please remove an extra space after '}'

File: install/install_openadapt.ps1 [hunk 2]
```diff
@@ -55,8 +59,7 @@ function RunAndCheck {
 function Cleanup {
     $exists = Test-Path -Path $setupdir
```
Comment (KrishPatel13) [inline install/install_openadapt.ps1:60]: Should not this line be changed to:

$exists = Test-Path -Path $openAdaptPath

Comment (KrishPatel13) [inline install/install_openadapt.ps1:60]: Should not this line be changed to:

`$exists = Test-Path -Path $openAdaptPath`

?

File: install/install_openadapt.ps1 [hunk 3]
```diff
@@ -55,8 +59,7 @@ function RunAndCheck {
...
     if ($exists) {
-        Set-Location $setupdir
-        Set-Location ../
+        Set-Location $env:USERPROFILE
         Remove-Item -LiteralPath $setupdir -Force -Recurse
```
Comment (KrishPatel13) [inline install/install_openadapt.ps1:63]: I see you used $openAdaptPath as a new setup dir where we will run the OpenAdapt's installation.

Should not this line be changed to:
`Remove-Item -LiteralPath $openAdaptPath -Force -Recurse` ?


Also on L298 and L299 we have a set-location to $setupdir. Should we change that to to $openAdaptPath as well ?

File: install/install_openadapt.ps1 [hunk 4]
```diff
@@ -55,8 +59,7 @@ function RunAndCheck {
 function Cleanup {
     $exists = Test-Path -Path $setupdir
     if ($exists) {
-        Set-Location $setupdir
-        Set-Location ../
+        Set-Location $env:USERPROFILE
```
Comment (KrishPatel13) [inline install/install_openadapt.ps1:62]: Should not this be: 

`Set-Location $openAdaptPath` ?

**Review by KrishPatel13 — CHANGES_REQUESTED** [review]
Some comments to address :-)

And then if possible could you record running the installation script on a Fresh VM and post a link to the media file here. :)

File: install/install_openadapt.ps1 [hunk 5]
```diff
@@ -4,21 +4,25 @@
...
+$tesseractInstallerURL = "https://digi.bib.uni-mannheim.de/tesseract/tesseract-ocr-w64-setup-5.3.1.20230401.exe"
 $tesseractPath = "C:\Program Files\Tesseract-OCR"
 
 $pythonCmd = "python"
 $pythonMinVersion = "3.10.0" # Change this if a different Lower version are supported by OpenAdapt
 $pythonMaxVersion = "3.10.12" # Change this if a different Higher version are supported by OpenAdapt
 $pythonInstaller = "python-3.10.11-amd64.exe"
-$pythonInstallerLoc = "https://www.python.org/ftp/python/3.10.11/python-3.10.11-amd64.exe"
+$pythonInstallerURL = "https://www.python.org/ftp/python/3.10.11/python-3.10.11-amd64.exe"
+# $pythonPath = "C:\Program Files\Python310;C:\Program Files\Python310\Scripts"
```
[the remark this answers is not in the document]
Comment (ayewo) [reply install/install_openadapt.ps1:20]: @KrishPatel13 sure.

File: install/install_openadapt.ps1 (hunk 2 above)
Comment (ayewo) [reply install/install_openadapt.ps1:60]: @KrishPatel13 
If we change that variable to `$openAdaptPath` as you are suggesting, which BTW is currently set to `"$env:USERPROFILE\OpenAdapt"` then it means that the installer will delete a non-existent `C:\User\<user>\OpenAdapt` folder which it hasn't used yet (which is fine), while trying to install dependencies i.e. the first 3 items in [this table](https://github.com/OpenAdaptAI/OpenAdapt/issues/447#issue-1829576911).

File: install/install_openadapt.ps1 (hunk 3 above)
Comment (ayewo) [reply install/install_openadapt.ps1:63]: My intention behind the changes is to use separate folders for separate tasks.
My proposal is for:
- the download and installation of dependencies to happen from `$setupdir`;
- the git clone and installation of OpenAdapt to happen from `$openAdaptPath`.

So in theory, someone who has the correct versions of all 3 dependencies (Tesseract, Python and Git) already installed would only have the install script clone and setup OpenAdapt from git.

Someone who has 2 out of the 3 dependencies would have the missing dep. installed by the script.

Someone who has the unsupported version of Python would have the correct version installed etc.

File: install/install_openadapt.ps1 (hunk 4 above)
Comment (ayewo) [reply install/install_openadapt.ps1:62]: Depending on where `Cleanup()` is called from, `$openAdaptPath` might not exist yet, which is why I used `$env:USERPROFILE` here as it is guaranteed to already exist.

Please see my additional clarification [above](https://github.com/OpenAdaptAI/OpenAdapt/pull/449#discussion_r1283464642).

Comment (KrishPatel13) [reply install/install_openadapt.ps1:61]: Thank you @ayewo 

The idea presented in the table looks great :)

File: install/install_openadapt.ps1 (hunk 2 above)
Comment (KrishPatel13) [reply install/install_openadapt.ps1:59]: 👍

**Comment by KrishPatel13** [comment]
Just for testing purpose

Run this command:

```
Start-Process powershell -Verb RunAs -ArgumentList '-NoExit', '-ExecutionPolicy', 'Bypass', '-Command', "iwr -UseBasicParsing -Uri 'https://raw.githubusercontent.com/OpenAdaptAI/OpenAdapt/cadff486af1f85cad84ffeb7c139984f0e49175a/install/install_openadapt.ps1' | Invoke-Expression"
```

**Comment by KrishPatel13** [comment]
@ayewo I tested the install script on this branch on fresh new Virtual Box (Windows 10): Here are some screenshots

image

**Comment by KrishPatel13** [comment]
```
<html><body><p>Drag and drop operation from guest to host failed.</p><qt><p>DnD: Error: Dragging from guest to host not supported by guest -- make sure that the Guest Additions are properly installed and running.</p>

Result Code: | VBOX_E_DND_ERROR (0X80BB0011)
-- | --
Component: | GuestDnDSourceWrap
Interface: | IGuestDnDSource {dedfb5d9-4c1b-edf7-fdf3-c1be6827dc28}
Callee: | IDnDSource {d23a9ca3-42da-c94b-8aec-21968e08355d}

<br></body></html><!--EndFragment-->
```

**Comment by KrishPatel13** [comment]
image

**Comment by KrishPatel13** [comment]
image

**Comment by KrishPatel13** [comment]
image

**Comment by KrishPatel13** [comment]
image

File: install/install_openadapt.ps1 [hunk 6]
```diff
@@ -181,15 +192,15 @@ function GetPythonCMD {
         $res = Invoke-Expression "python -V"
         $versionString = $res.Split(' ')[-1]
```
Comment (KrishPatel13) [inline install/install_openadapt.ps1:193]: Looks like there is some issue related to this line. (Noticed form 1st Screenshot)

Comment (ayewo) [reply install/install_openadapt.ps1:193]: Hmm, I didn't touch the lines you referenced here, at all.

**Comment by ayewo** [comment]
> image

This screenshot provides a hint as to why the tests might be failing in your fresh Windows environment:
> Microsoft Visual C++ Redistributable is not installed, this may lead to the DLL load failure.
It can be downloaded at [https://aka.ms/vs/16/release/vc_redist.×64.exe](https://aka.ms/vs/16/release/vc_redist.x64.exe)

**Comment by KrishPatel13** [comment]
Great Comments :-)

@ayewo, The above comment is very helpful to us. Thank you :-)

We had removed the VS Redist installation from the file (unknowing why it was added earlier).

https://github.com/OpenAdaptAI/OpenAdapt/pull/449#issuecomment-1669999300
https://github.com/OpenAdaptAI/OpenAdapt/pull/449#issuecomment-1670001954

But, now the above screenshot proves that we will indeed need it. We were waiting for this (so that we could reproduce an error for VS C++ Redist).

Now, let us make an separate issue that just add VS C++ Redist installation to the current main branch installation script and then resume the testing for this branch once that issue is merged.

<Add Link to new Issue for Adding VS C++ installation in Powershell script>

Please let me know your thoughts @ayewo and @abrichr 

:-)

File: install/install_openadapt.ps1 (hunk 6 above)
Comment (KrishPatel13) [reply install/install_openadapt.ps1:193]: Yes you are right, but we will have to fix it :)
Let us fix that in this pr only. If that is fine. :-)

**Comment by ayewo** [comment]
Hey @KrishPatel13 

Sorry I've not been able to respond.

I see it's been merged, thanks!

**Comment by KrishPatel13** [comment]
It is completely fine. Let us finish this weekend if you are free :)

I have also loged the PR for adding Visual Studio C++ Redist.

**Comment by abrichr** [comment]
@ayewo @KrishPatel13 https://github.com/OpenAdaptAI/OpenAdapt/pull/490 has been merged. What's left here?

**Comment by ayewo** [comment]
@abrichr not sure either. 

Perhaps @KrishPatel13 can comment?

**Comment by ayewo** [comment]
Oops! Sorry.

(I unintentionally closed this PR when I renamed the branch from `main` to [`fix-windows-installer`](https://github.com/OpenAdaptAI/OpenAdapt/compare/main...ayewo:OpenAdapt:fix-windows-installer.)

It looks like my mistake can't be undone as I've synced my fork with `main` with a forced-push that deleted my 3 commits.

**Pull request closed without merging**